src/data/main.py

```diff
@@ -1,9 +1,248 @@
+import os
+import sys
+import yaml
+import pandas as pd
+
+from click import MissingParameter
+from searchtweets import gen_request_parameters, load_credentials, collect_results
+
 from src.data.make_dataset.aclImdb import make_dataset_aclImdb
-from src.data.make_dataset.macron import make_dataset_macron
+
+# ########################################################################### #
+#                                 Constants                                   #
+# ########################################################################### #
+EXPECTED_KEYS_CREDENTIALS = ['SEARCHTWEETS_ENDPOINT',
+                             'SEARCHTWEETS_BEARER_TOKEN',
+                             'SEARCHTWEETS_CONSUMER_KEY',
+                             'SEARCHTWEETS_CONSUMER_SECRET']
+
+CANDIDATS = ["Valérie Pécresse",
+             "Eric Zemmour",
+             "Nicolas Dupont-Aignan",
+             "Jean-Luc Mélenchon",
+             "Marine Le Pen",
+             "Jean Lassalle",
+             "Anne Hidalgo",
+             "Emmanuel Macron",
+             "Yannick Jadot",
+             "Fabien Roussel",
+             "Nathalie Arthaud",
+             "Philippe Poutou"]
+
+NOMS = ["Pecresse",
+        "Zemmour",
+        "Dupont-Aignan",
+        "Melenchon",
+        "Le Pen",
+        "Lassalle",
+        "Hidalgo",
+        "Macron",
+        "Jadot",
+        "Roussel",
+        "Arthaud",
+        "Poutou"]
+
+TAGS = {"Pecresse": "@avecValerie",
+        "Zemmour": "@ZemmourEric",
+        "Dupont-Aignan": "@dupontaignan",
+        "Melenchon": "@JLMelenchon",
+        "Le Pen": "@MLP_officiel",
+        "Lassalle": "@jeanlassalle",
+        "Hidalgo": "@Anne_Hidalgo",
+        "Macron": "@EmmanuelMacron",
+        "Jadot": "@yjadot",
+        "Roussel": "@AvecRoussel",
+        "Arthaud": "@n_arthaud",
+        "Poutou": "@PhilippePoutou"}
+
+CREDENTIAL_FILE = '../../.twitter_keys.yaml '
+# maybe later we would be able to use all, then we could change with search_tweets_v2_all
+TWITTER_KEY = 'search_tweets_v2_recent'
+NB_MAX_TWEETS = 40  # Max number of tweets one wants
+RES_PER_CALL = NB_MAX_TWEETS  # Max number of tweets per query to Twitter API
+CONFIG_YAML = ".config.yaml"
+SAVE_PATH = "data/raw/twitter"
+# ########################################################################### #
+#                                 Functions                                   #
+# ########################################################################### #
+
+
+def dataset_to_csv(df_dataset: pd.DataFrame, filename: str):
+    """ Saves the dataset into a file in the specified directory.
+    Args:
+    -----
+        df_dataset [pandas DataFrame]: dataframe containing the dataset.
+        filename [str]: name of the file where df will be saved.
+    Return:
+    -------
+        None
+    """
+    csv_path = f"{SAVE_PATH}/{filename}.csv"
```

Comment:
Typo: double .csv aplied with line 240. Fixed in commit 533e7b6d6c387d523d90541a795a692d893de5ac